### PR TITLE
Refactor logic around `winning_bid`

### PR DESCRIPTION
**Pull request by jessieay**
* This represents business logic and therefore makes sense to include in
  `auction`
* Related `status` logic was hard to parse, moved that into service
  class

### CODE REVIEW

File: app/models/auction.rb [hunk 1]
```diff
@@ -1,11 +1,29 @@
 class Auction < ActiveRecord::Base
   has_many :bids
   has_many :bidders, through: :bids
-  enum result: {not_applicable: 0, accepted: 1, rejected: 2}
-  enum type: {single_bid: 0, multi_bid: 1}
-  enum awardee_paid_status: {not_paid: 0, paid: 1}
-  enum published: {unpublished: 0, published: 1}
+  enum result: { not_applicable: 0, accepted: 1, rejected: 2 }
```
Comment (harrisj) [inline app/models/auction.rb]: I would like to convert these from enums to strings, but not urgent. Just noting this is a vague to-do of mine

File: app/models/auction.rb [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
   # Disable STI
   self.inheritance_column = :__disabled
+
+  def winning_bid
+    if single_bid? && AuctionStatus.new(self).available?
+      nil
+    else
+      lowest_bid
+    end
+  end
```
Comment (harrisj) [inline app/models/auction.rb]: I don't think winning_bid should be in the AR model because it depends on the auction type and I'd rather keep basic methods only in here

**Comment by harrisj** [comment]
I feel like this and #419 are trying to do similar things and I'm not sure if we should hash out some hybrid approach between the two

File: app/models/auction.rb (hunk 1 above)
Comment (jessieay) [inline app/models/auction.rb]: I like them as enums (get free methods like `not_applicable?`) -- what do you see as the benefit of strings?

File: app/models/auction.rb (hunk 2 above)
Comment (jessieay) [inline app/models/auction.rb]: I think of `winning_bid` as a basic model -- this app at a basic level holds auctions and determines a winner, so I think elevating winner beyond view logic makes sense. We can move out of `auction`, but let's discuss !

**Comment by jessieay** [comment]
I think we can def do a hybrid! What I am seeing in 419 is a breaking up of how bids are placed for single vs multi bids. I think we can also do that for how we determine winning bids. The main point of this PR was to highlight that `winning_bid` is as business rules concept, not something purely for the view / decorator. In the process of seeing how the presenter determined `winning_bid`, I saw all of the `status` stuff around auctions and decided that could go in its own service class, so that became part of this as well.

**Comment by harrisj** [comment]
Looks good to me. I want to move `winning_bid` to Rule objects as in #419 eventually, but this is needed for a feature to be implemented and I don't want to block that.
